grails-app/controllers/com/unifina/controller/StorageNodeApiController.groovy

```diff
@@ -0,0 +1,75 @@
+package com.unifina.controller
+
+import com.unifina.security.StreamrApi
+import com.unifina.security.AuthLevel
+import com.unifina.domain.Stream
+import com.unifina.domain.StreamStorageNode
+import com.unifina.service.StorageNodeService
+import com.unifina.service.PermissionService
+import com.unifina.api.ValidationException
+import com.unifina.api.NotPermittedException
+import com.unifina.utils.EthereumAddressValidator
+import com.unifina.domain.Permission
+import com.unifina.domain.User
+import grails.converters.JSON
+import grails.validation.Validateable
+import org.apache.log4j.Logger
+
+@Validateable
+class StorageNodeAddCommand {
+	String address
+
+	static constraints = {
+		address(nullable: false, validator: EthereumAddressValidator.validate)
+	}
+}
+
+class StorageNodeApiController {
+
+	private static final Logger log = Logger.getLogger(StorageNodeApiController)
+
+	StorageNodeService storageNodeService
+	PermissionService permissionService
+
+	@StreamrApi(authenticationLevel = AuthLevel.NONE)
+	def findStreamsByStorageNode(String storageNodeAddress) {
+		List<Stream> streams = storageNodeService.findStreamsByStorageNode(storageNodeAddress);
+		return render(streams*.toSummaryMap() as JSON)
+	}
+
+	@StreamrApi(authenticationLevel = AuthLevel.NONE)
+	def findStorageNodesByStream(String streamId) {
```

Comment:
`streamId` requires validation.

Reply:
Let's add validation to streamId once it has specific format (hex-string or ens path)